science/gnuradio/Portfile

```diff
@@ -526,276 +367,209 @@ variant qtgui description "Install GNU Radio with support for Qt GUI" {
-
         depends_run-append \
-            port:py${PythonVersionNoDot}-pyqwt
+            port:py${active_python_version_no_dot}-pyqwt
 
         configure.args-append \
             -DQWT_INCLUDE_DIRS=${prefix}/include/qwt \
             -DQWT_LIBRARIES=${prefix}/lib/libqwt.dylib
 
     } else {
 
-        # next
         PortGroup qt5 1.0
 
         # pyqt5 is checked for at configure, then required for
         # runtime; so use depends_lib to get both.  Can use
         # either py*-pyqt5 or py*-pyqt5-devel.
-
         depends_lib-append \
-            path:share/py${PythonVersionNoDot}-sip/PyQt5:py${PythonVersionNoDot}-pyqt5
+            path:share/py${active_python_version_no_dot}-sip/PyQt5:py${active_python_version_no_dot}-pyqt5
 
         # Requires qwt >= 6.0, installed for qt5.
         # just use the new port that does this.
-
         depends_lib-append port:qwt-qt5
 
+        # require PyQtGraph for runtime only
+        depends_run-append port:py${active_python_version_no_dot}-pyqtgraph
+
         configure.args-append \
             -DQWT_INCLUDE_DIRS=${qt_dir}/lib/qwt.framework/Versions/6/Headers \
             -DQWT_LIBRARIES=${qt_dir}/lib/qwt.framework/qwt
 
-        # require PyQtGraph for runtime only
-        depends_run-append port:py${PythonVersionNoDot}-pyqtgraph
-
     }
 
+    configure.args-delete \
+        -DENABLE_GR_QTGUI=OFF \
+        -DQT_QMAKE_EXECUTABLE=
+
     configure.args-append \
         -DENABLE_GR_QTGUI=ON \
         -DQT_QMAKE_EXECUTABLE=${qt_qmake_cmd}
 
 }
 
-if {![variant_isset qtgui]} {
 
-    configure.args-append \
-        -DENABLE_GR_QTGUI=OFF \
-        -DQT_QMAKE_EXECUTABLE=
 
-}
+if {${subport} eq "gnuradio37"} {
+    variant wxgui description "Install GNU Radio with support for Wx GUI" {
 
-if {${subport} eq "gnuradio-next"} {
-    # no wxgui in next!
-} else {
+        PortGroup wxWidgets 1.0
 
-    variant wxgui description "Install GNU Radio with support for Wx GUI" {
+        wxWidgets.use wxWidgets-3.0
 
         depends_lib-append \
             port:${wxWidgets.port}
 
         # wxpython is checked for at configure, then required
         # for runtime; so use depends_lib to get both.
-
         depends_lib-append \
-            port:py${PythonVersionNoDot}-wxpython-3.0
+            port:py${active_python_version_no_dot}-wxpython-3.0
 
         configure.args-append \
-            -DENABLE_GR_WXGUI=ON
-
-    }
-
-    if {[variant_isset wxgui]} {
-
-        PortGroup wxWidgets 1.0
-
-        wxWidgets.use wxPython-3.0
-
-    } else {
+            -DENABLE_GR_WXGUI=OFF
 
         configure.args-append \
-            -DENABLE_GR_WXGUI=OFF
+            -DENABLE_GR_WXGUI=ON \
+            -DwxWidgets_CONFIG_EXECUTABLE=${wxWidgets.wxconfig}
 
     }
 }
 
 variant uhd description "Install GNU Radio with support for UHD" {
 
     # allow for uhd or uhd-devel
-
     depends_lib-append \
         path:lib/libuhd.dylib:uhd
 
+    configure.args-append \
```

Comment:
not here. this code is for when the variant is -not- set. I'd say to leave the prior code alone.

Reply:
oh .. you probably meant to add in the above to the top-level configure.args-append, then `delete` it here... that's fine ... whichever works for you! just not `append` here!

Reply:
as always, copy&paste is the evil :)